### DISCUSSION TITLE
Add a "contain" option in the search box

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#1421] [Anything Else]
The "Add Filter" search box currently has "is" and "is not" options. A "contain" option would be great

**Update:** In the filter menu and in pageview goals, you can use double asterisks in front and back of your keyword (`**keyword**`) in the same way that you would use contain. It finds any URL containing your specific keyword.

**Update 2:** Contains filter type for page-based metrics is now available

**Comment by andreamazza89** [comment]
Hi 👋 I've started looking into this, and want to make sure I understand correctly.

I am guessing that "contain" would be doing a loose match, so for example if I set up a filter like this:

`Page` `contains` `xample`

Then it would match on pages like `/Example`, `/Example1`, `/some-other-route/MyExample`

Two questions:

1. Is the above correct - is it what you had in mind?

2. If so, do we want this concept to apply to filters that are not path-based? In other words, does it make sense to use this kind of filtering for things like `Browser`, `Goal`, `OS`?

**Reply by metmarkosaric** [reply]
hi @andreamazza89! 

yes, that's exactly what we had in mind for this. it's most relevant for pages, entry pages and exit pages. i can see a use for it with UTM tags too, perhaps referral sources and goals as well. 

for countries/regions/cities and for devices, browsers and OS i don't think it's necessary as i'm not sure that it can be useful in any situation for these 

thanks!

**Reply by andreamazza89** [reply]
Awesome, thank you for your reply @metmarkosaric 👍

At the moment I think we can already achieve this using double asterisks in front as well as after the term; for example, [this filter](https://plausible.io/plausible.io?page=**api**) (using `**api**`) matches on `/docs/stats-api`, `/settings/api-keys/new` and more.

So think there are two options:

1. Keeping as is (i.e. you still select the `is` criterion on the filter and use `**` for loose matches) and enabling "asterisk-based" filters on UTM tags, perhaps referral sources and goals.
2. Adding a `contains` criterion on the filter, which effectively wraps the filter term in asterisks (i.e. `**<filter term>**`). This option (`contains`) should only be available for certain filters. I wonder if this option might be confusing as we are offering two ways of doing the same thing?

**Reply by metmarkosaric** [reply]
oh cool! i wasn't aware that `**api** ` can be used like "contain". in that case, perhaps number 2 is best. we just need to document this and add a hint like we have the hint for double asterisks

**Reply by ukutaht** [reply]
My 2 cents: the asterisk thing was great for developers who are used to glob, regexes and stuff like that. But non-devs definitely have a harder time using this feature. I think it would be useful to add a `contains` filter type because it's easier to use for most people than figuring out how to asterisks work.

I'm not worried about offering multiple ways to do the same thing here. Even devs will probably use `contains` most of the time and only use the asterisk method when more powerful filtering is necessary.

I also agree that it doesn't make a lot of sense for countries, browsers, etc. The `contains` filter type is useful for filtering `page`, `entry_page` and `exit_page`.

**Reply by andreamazza89** [reply]
Thank you for that :+1: - will add a `contains` option for `page` `entry_page` and `exit_page`

**Reply by metmarkosaric** [reply]
Thanks for your help! Might be worth combining pages, entry pages and exit pages as one item in the filter menu as well rather than having them as three different items. Not sure if that adds a lot of extra work

**Reply by andreamazza89** [reply]
Here it is 😄 :

https://github.com/plausible/analytics/pull/1799
https://github.com/plausible/docs/pull/206

**Reply by metmarkosaric** [reply]
nice!

**Comment by metmarkosaric** [comment]
A bit of feedback after we announced "contains" today:

"Would love this for Sources too" by [Nico](https://twitter.com/distantnative/status/1509181678866022407)

And:

"Is it going to be available in the API as well?" by [Gustavo](https://twitter.com/theluctus/status/1509174794348027908) (related thread https://github.com/plausible/analytics/discussions/1282)

**Reply by andreamazza89** [reply]
cool!

Adding 'contains' to Sources is very easy, so do let me know if you'd like me to do it.

Regarding the API, that's a bit more involved, so will not jump on it right away, but good to know 👍

**Reply by metmarkosaric** [reply]
Yeah, I think it would be best to add it to sources as well (perhaps even to utm tags and goals). I see you're working on the search box for /sites next and that one is more of a priority as it's asked for more frequently.

**Comment by webtail** [comment]
Can the 'contain' option also be added to the utm's as suggested here above :)?

**Reply by webtail** [reply]
No idea if the 'contain' value can also be implemented with the utm filters @metmarkosaric ? THxs!

**Comment by metmarkosaric** [comment]
**Update related to this**: In addition to the "contain" for pages, you can now also filter your stats by multiple countries, pages, sources and more, all at once. Click on the "Filter" button in your dashboard to start exploring.

**Reply by webtail** [reply]
Thxs @metmarkosaric . But no 'contain' for utm parameters right?

**Reply by metmarkosaric** [reply]
hi @webtail! that's right. the "contains" is only for pages. for UTMs you can now combine multiple UTMs using the filter button

**Reply by webtail** [reply]
Yes saw that, issue is that sometimes it's very handy to see/filter certain campaigns together. Being able to say: show me all campaigns that contains XYZ would be very handy :)

**Reply by metmarkosaric** [reply]
i understand. for now you can manually add all the campaigns together using the new filter. not as handy as "contains" but it does the job too. we'll keep an eye on the popularity of the idea to combine utms using "contains" and will update in case we have any progress

**Reply by metmarkosaric** [reply]
hi @webtail! "Contains" is now available for UTMs too!

**Reply by webtail** [reply]
wow thxs!!!
